src/Microsoft.DotNet.CodeFormatting/Analyzers/ExplicitVariableTypeFixer.cs

```diff
@@ -0,0 +1,154 @@
+            => ImmutableArray.Create(ExplicitVariableTypeAnalyzer.DiagnosticId);
+
+        public override FixAllProvider GetFixAllProvider()
+        {
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            CancellationToken cancellationToken = context.CancellationToken;
+            SyntaxNode root = await context.Document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            SemanticModel model = await context.Document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            Diagnostic diagnostic = context.Diagnostics.First();
+            TextSpan diagnosticSpan = diagnostic.Location.SourceSpan;
+
+            TypeSyntax oldTypeSyntaxNode = null;
+            TypeSyntax newTypeSyntaxNode = null;
+            switch (GetRuleType(diagnostic))
+            {
+                case RuleType.RuleVariableDeclaration:
+                    var variableDeclarationNode = root.
+                        FindNode(diagnosticSpan).
+                        FirstAncestorOrSelf<VariableDeclarationSyntax>();
+                    Debug.Assert(variableDeclarationNode != null);
+                    oldTypeSyntaxNode = variableDeclarationNode.Type;
+                    // Implicit typed variables cannot have multiple declartors
+                    newTypeSyntaxNode = CreateTypeSyntaxNode(variableDeclarationNode.Variables.Single(), model, cancellationToken);
+                    break;
+
+                case RuleType.RuleForEachStatement:
+                    var forEachStatementNode = root.
+                        FindToken(diagnosticSpan.Start).
+                        Parent.
+                        FirstAncestorOrSelf<ForEachStatementSyntax>();
+                    Debug.Assert(forEachStatementNode != null);
+                    oldTypeSyntaxNode = forEachStatementNode.Type;
+                    newTypeSyntaxNode = CreateTypeSyntaxNode(forEachStatementNode, model, cancellationToken);
+                    break;
+
+                case RuleType.None:
+                    break;
+            }
+
+            if (oldTypeSyntaxNode != null && newTypeSyntaxNode != null)
+            {
+                context.RegisterCodeFix(
+                    CodeAction.Create(
+                        Resources.ExplicitVariableTypeFixer_Title,
+                        c => ReplaceVarWithExplicitType(context.Document, root, oldTypeSyntaxNode, newTypeSyntaxNode.WithTriviaFrom(oldTypeSyntaxNode))),
+                    diagnostic);
+            }
+        }
+
+        private Task<Document> ReplaceVarWithExplicitType(Document document, SyntaxNode root, SyntaxNode varNode, SyntaxNode explicitTypeNode)
+        {
+            return Task.FromResult(
+                document.WithSyntaxRoot(root.ReplaceNode(varNode, explicitTypeNode.WithAdditionalAnnotations(Simplifier.Annotation))));
```

Comment:
@jaredpar @Priya91 So this doesn't simplify system types like `Int32` and `String` to keyword `int` and `string`. Did I do anything wrong here?